gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/map/SubstringStep.java

```diff
@@ -104,4 +104,15 @@ private int processStartIndex(int strLen) {
         }
     }
 
+    // Helper function to process the end index, if it is negative (which counts from end of string) it is converted
+    // to the positive index position or 0 when negative index exceeds the string length, which means an empty string
+    // will be returned, if it is positive and exceeds the length of the string, it is converted to the length of the
+    // string.
+    private int processEndIndex(int strLen) {
```

Comment:
This is basically the same function as processStartIndex(). Consider refactoring.

Reply:
Merged into processStringIndex()